### PR TITLE
Segmentation

### PR DESCRIPTION
**Pull request by stefanv**

### CODE REVIEW

File: tutorial/03_segmentation_and_measuring.md [hunk 1]
```diff
@@ -0,0 +1,726 @@
...
+
+--------------
+
+One can think of an image as a projection of a scene, consisting of multiple objects. Sometimes, the scene is considered as a whole, but quite often it is useful to identify the constituent parts, or segments. E.g., we may want to count those segments, or calculate some properties of them. E.g., the number and sizes of cells in a microscopy slide, or the positions and poses of people in a photograph.
+
+Splitting an image into its constituent parts is known as *segmentation*.
+
+Think of Michaelangelo, carving the statue of David. The block of marble is *segmented* into foreground and background. In this case, the background is discarded, and only the work of art remains.
+
+To be useful, segmentation has to be followed by *measurement* and *interpretation*. We measure attributes of the segments, and then interpret those measurements to ask some question. Ultimately, we want to know: are the cells healthy, or is the part manufactured correctly, or is the land flooded.
```
Comment (mkcor) [inline tutorial/03_segmentation_and_measuring.md:40]:
```suggestion
To be useful, segmentation has to be followed by *measurement* and *interpretation*. We measure attributes of the segments, and then interpret those measurements to answer some question. Ultimately, we want to know: are the cells healthy, or is the part manufactured correctly, or is the land flooded.
```
or 'try and answer'

File: tutorial/03_segmentation_and_measuring.md [hunk 2]
```diff
@@ -0,0 +1,726 @@
...
+flowchart LR
+
+A[Imaging] --> B[Segmentation]
+B --> C[Measurement]
+C --> D[Interpretation]
+```
+
+Let's consider that second block, the segmentation algorithm, for a moment.
+
+Scikit-image implements several *heuristic* algorithms for segmentation. I.e., these are typically *unsupervised*, based on rules around properties of the image / pixels, rather than on labeling examples of objects and non-objects. Neural network approaches, such as U-Net, DeepLab, and Mask R-CNN have proved very effective for segmenting images, given enough training data. These are likely what would be used in practice, unless sufficient labeled data is unavailable.
```
Comment (mkcor) [inline tutorial/03_segmentation_and_measuring.md]: The double negative is twisting my brains... "unless not enough labeled data are available?"

File: tutorial/03_segmentation_and_measuring.md [hunk 3]
```diff
@@ -0,0 +1,726 @@
...
+
+In terms of this tutorial, we will show the pipeline above, with the understanding that the segmentation algorithm can always be swapped around for something more sophisticated. The results still have to be analysed.
+
++++
+
+### Representing segmentations
+
+How shall we go about representing segmentations? We already know how to use `numpy` to represent images. Can we use the same concept here?
+
+Yes, we can create a *label* image, of the same size as the segmented image, with each value getting a different value.
```
Comment (mkcor) [inline tutorial/03_segmentation_and_measuring.md:62]:
```suggestion
Yes, we can create a *label* image, of the same shape as the segmented image, with each segment getting a different value.
```
Maybe a "labels" image/array? E.g., see
from typing import Annotated
```py
ImageData = Annotated[np.ndarray, 'image']
LabelsData = Annotated[np.ndarray, 'labels']
CoordsData = Annotated[np.ndarray, 'coordinates']
```
from https://github.com/scikit-image/skimage-archive/blob/main/grants/2022_CZI_EOSS5/full_application.md

File: tutorial/03_segmentation_and_measuring.md [hunk 4]
```diff
@@ -0,0 +1,726 @@
...
+ax[0].imshow(image)
+ax[0].set_title('image')
+ax[1].imshow(-distance, cmap='gray')
+ax[1].set_title('negative\ndistance transform')
+ax[1].scatter(coords[:, 1], coords[:, 0], marker='x', s=20, c='red')
+ax[2].imshow(labels)
+ax[2].set_title('labels');
+```
+
+Notice that "labels" is just a NumPy array with integer values. We have to be careful to interpret it as labels and not as an image.
```
Comment (mkcor) [inline tutorial/03_segmentation_and_measuring.md:187]: See, "labels."
